mitodl/unified-ecommerce
# Updates to the backend to support the frontend

### What are the relevant tickets?

https://github.com/mitodl/hq/issues/5656

### Description (What does it do?)

Changes some stuff to make things work better for the frontend code:
- Expose the CSRF settings so we can set them
- Fixes the API spec for the add discount API to add in the discount code parameter (wasn't there before)
- Exposes the `start_checkout` API to the OpenAPI spec
- Include discount information with basket items
- Reworks the `start_checkout` API to be a view function so it's easier to route and generate docs for
- Regenerate the OpenAPI specs/clients

### How can this be tested?

Tests should pass. You should see the start checkout API in Swagger, you should be able to see the discount code property the add discount API, and baskets should include discount information if there is any for the items within.

This branch is necessary to make [this PR in the frontend repo](https://github.com/mitodl/unified-ecommerce-frontend/pull/10) work, so you can test by checking that out and then using the frontend.
